src/client.rs

```diff
@@ -67,13 +62,8 @@ impl<F: FieldElement> Client<F> {
         }
```

Comment:
Seems like this check should move down to `ClientMemory::new` since `n` isn't used in this function.